sky/clouds/service_catalog/common.py

```diff
@@ -270,6 +270,25 @@ def get_vcpus_from_instance_type_impl(
     return float(vcpus)
 
 
+def _filter_with_cpu(df: pd.DataFrame, cpu: Optional[str]) -> pd.DataFrame:
+    if cpu is None:
+        return df
+    if cpu.endswith('+'):
+        return df[df['vCPUs'] >= float(cpu[:-1])]
+    else:
+        return df[df['vCPUs'] == float(cpu)]
+
+
+def get_default_instance_type_impl(df: pd.DataFrame,
+                                   cpu: Optional[str] = None) -> Optional[str]:
+    df = _filter_with_cpu(df, cpu)
+    if df.empty:
+        return None
+    # Sort by the number of vCPUs and then by the price.
+    df = df.sort_values(by=['vCPUs', 'Price'], ascending=True)
+    return df['InstanceType'].iloc[0]
```

Comment:
See comment in aws.py. This feels more like filtering vs. getting the default?

Reply:
For this function, I agree that it has nothing to do with "default". Renamed to `get_instance_type_for_cpu_impl`.